### PR TITLE
SWL-4652 Add tlv and enums for port_mode.

### PR DESCRIPTION
**Pull request by wilmo119**
Reviewer: @poolakiran @andi-bigswitch

### CODE REVIEW

**Comment by bsn-abat** [comment]
Find artifact changes from this pull request at https://github.com/floodlight/loxigen-artifacts/tree/PR-622-b1

**Comment by andi-bigswitch** [comment]
LGTM!

File: openflow_input/bsn_tlv [hunk 1]
```diff
@@ -1305,3 +1305,22 @@ struct of_bsn_tlv_lr_all_enabled : of_bsn_tlv {
     uint16_t type == 178;
     uint16_t length;
 };
+
+enum ofp_bsn_port_mode(wire_type=uint16_t) {
+    OFP_BSN_PORT_MODE_4XX = 0,
```
Comment (poolakiran) [inline openflow_input/bsn_tlv]: Can we add OFP_BSN_PORT_MODE_NONE = 0, which can be used for default assignment. And also it helps if it is default value or controller configured value. Controller should never use OFP_BSN_PORT_MODE_NONE.

**Comment by poolakiran** [comment]
one minor comment, otherwise lgtm. please self-ack when ready.

**Comment by bsn-abat** [comment]
Find artifact changes from this pull request at https://github.com/floodlight/loxigen-artifacts/tree/PR-622-b2

**Comment by wilmo119** [comment]
SELF-ACK

**Comment by bsn-abat** [comment]
ABAT: START_MERGE:  https://jenkins.bigswitch.com/job/loxigen-abat-automerge/12935//console

**Comment by bsn-abat** [comment]
ABAT: ACCEPT:  Successfully merged

In case you want to see the build log, check out :
https://jenkins.bigswitch.com/job/loxigen-abat-automerge/12935//console
/cc

**Pull request closed without merging**